python/packages/autogen-core/src/autogen_core/code_executor/_func_with_reqs.py

```diff
@@ -152,14 +152,67 @@ def __call__(self, *args: P.args, **kwargs: P.kwargs) -> T:
 def with_requirements(
     python_packages: Sequence[str] = [], global_imports: Sequence[Import] = []
 ) -> Callable[[Callable[P, T]], FunctionWithRequirements[T, P]]:
-    """Decorate a function with package and import requirements
+    """
+    Decorate a function with package and import requirements for code execution environments.
```

Comment:
We can note which code executors currently support this: 

1. DockerCommandLineCodeExecutor
2. LocalCommandLineCodeExecutor
3. ACADynamicSessionCodeExecutor